DYDX-Bots/Insolvent_Contract/src/balance.fetcher.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { createAddress, MockEthersProvider } from "forta-agent-tools/lib/tests";
+import { BigNumber } from "ethers";
+import BalanceFetcher from "./balance.fetcher";
+import { IMPLEMENTATION_IFACE } from "./utils";
+
+describe("Balance Fetcher test suite", () => {
+  const testProxyAddress: string = createAddress("0xab");
+  // Format: [totalBorrowerDebtBalance, totalActiveBalanceCurrentEpoch, blockNumber]
+  const TEST_CASES: [BigNumber, BigNumber, number][] = [
+    [BigNumber.from("10"), BigNumber.from("80"), 1],
+    [BigNumber.from("20"), BigNumber.from("85"), 2],
+    [BigNumber.from("30"), BigNumber.from("90"), 3],
+    [BigNumber.from("40"), BigNumber.from("95"), 4],
+  ];
+
+  const mockProvider: MockEthersProvider = new MockEthersProvider();
+  const fetcher: BalanceFetcher = new BalanceFetcher(mockProvider as any, testProxyAddress);
+
+  function createGetTotalBorrowerDebtBalance(totalBorrowerDebtBalance: BigNumber, blockNumber: number) {
+    return mockProvider.addCallTo(testProxyAddress, blockNumber, IMPLEMENTATION_IFACE, "getTotalBorrowerDebtBalance", {
+      inputs: [],
+      outputs: [totalBorrowerDebtBalance],
+    });
+  }
+
+  function createGetTotalActiveBalanceCurrentEpoch(totalActiveBalanceCurrentEpoch: BigNumber, blockNumber: number) {
+    return mockProvider.addCallTo(
+      testProxyAddress,
+      blockNumber,
+      IMPLEMENTATION_IFACE,
+      "getTotalActiveBalanceCurrentEpoch",
+      {
+        inputs: [],
+        outputs: [totalActiveBalanceCurrentEpoch],
+      }
+    );
+  }
+
+  beforeEach(() => mockProvider.clear());
+
+  it("should store correct total borrower debt balance amount", async () => {
+    for (let [totalBorrowerDebtBalance, , blockNumber] of TEST_CASES) {
+      createGetTotalBorrowerDebtBalance(totalBorrowerDebtBalance, blockNumber);
+
+      const fetchedDebtBalance: BigNumber = await fetcher.getTotalBorrowerDebtBalance(blockNumber);
+      expect(fetchedDebtBalance).toStrictEqual(totalBorrowerDebtBalance);
+    }
+
+    // clear mock to use cache
+    mockProvider.clear();
+    for (let [totalBorrowerDebtBalance, , blockNumber] of TEST_CASES) {
+      createGetTotalBorrowerDebtBalance(totalBorrowerDebtBalance, blockNumber);
+
+      const fetchedDebtBalance: BigNumber = await fetcher.getTotalBorrowerDebtBalance(blockNumber);
+      expect(fetchedDebtBalance).toStrictEqual(totalBorrowerDebtBalance);
```

Comment:
In order to use cache, we don't need to call `createGetTotalBorrowerDebtBalance` again. Same note applies to the following test case.